services/src/validator.rs

```diff
@@ -647,20 +647,38 @@ Error: {err}",
             let network =
                 zebra_chain::parameters::Network::new_regtest(ConfiguredActivationHeights {
                     before_overwinter: Some(1),
-                    overwinter: Some(
-                        self.activation_heights
-                            .set_height(NetworkUpgrade::Overwinter),
-                    ),
-                    sapling: Some(self.activation_heights.set_height(NetworkUpgrade::Sapling)),
-                    blossom: Some(self.activation_heights.set_height(NetworkUpgrade::Blossom)),
-                    heartwood: Some(
-                        self.activation_heights
-                            .set_height(NetworkUpgrade::Heartwood),
-                    ),
-                    canopy: Some(self.activation_heights.set_height(NetworkUpgrade::Canopy)),
-                    nu5: Some(self.activation_heights.set_height(NetworkUpgrade::Nu5)),
-                    nu6: Some(self.activation_heights.set_height(NetworkUpgrade::Nu6)),
-                    nu6_1: Some(self.activation_heights.set_height(NetworkUpgrade::Nu6_1)),
+                    overwinter: self
```

Comment:
We should find a way to express this repetitive code more succinctly.   I misnamed the set_height method I defined..  should have been get_..   another option might be a macro.  There could be another solution, but we ought not to be repetitive if another option is feasible.   Repetition exposes us to bugs.

Reply:
the best option here is to have a try_into impl to convert from ActivationHeight to ConfiguredActivationHeights.. can i do this in a following PR so i unblock zingolib?